### PR TITLE
[Docs] Refresh "Contributing code" section

### PR DESCRIPTION
**Pull request by pamil**

### CODE REVIEW

File: docs/contributing/code/bc.rst [hunk 1]
```diff
@@ -59,12 +59,40 @@ All the new routes will start with ``sylius_`` prefix in order to avoid conflict
...
+A code is marked as deprecated by adding a ``@deprecated`` PHPDoc to
+relevant classes, methods, properties:
+
+.. code-block:: php
+
+    /**
+     * @deprecated Deprecated since version 1.X. Use XXX instead.
+     */
+
+The deprecation message should indicate the version when the class/method was
```
Comment (CoderMaggie) [inline docs/contributing/code/bc.rst]:
```suggestion
The deprecation message should indicate the version in which the class/method was
```

File: docs/contributing/code/security.rst [hunk 2]
```diff
@@ -1,47 +1,12 @@
...
-   * the possible exploits;
-   * how to patch/upgrade/workaround affected applications;
-   * credits.
-4. Send the patch and the announcement to the reporter for review;
-5. Apply the patch to all maintained versions of Sylius;
-6. Publish the post on the official Sylius `blog`_;
-7. Update the security advisory list (see below).
+If you think that you have found a security issue in Sylius, don't use the bug tracker and do not post it publicly.
+Instead, all security issues must be sent to **security@sylius.com**.
+Emails sent to this address are forwarded to the Sylius Core Team Members responsible for the framework security.
```
Comment (CoderMaggie) [inline docs/contributing/code/security.rst]:
```suggestion
Emails sent to this address are forwarded to the Sylius Core Team Members responsible for the framework's security.
```

File: docs/contributing/code/security.rst [hunk 3]
```diff
@@ -1,47 +1,12 @@
...
-    Sunday, except if the vulnerability has been publicly posted.
-
-.. note::
-
-    While we are working on a patch, please do not reveal the issue publicly.
-
-.. _Git repository:      https://github.com/Sylius/Sylius
-.. _blog:                http://sylius.com/blog
+    We will not publish security releases during Saturday or Sunday,
+    except if the vulnerability has been made public.
```
Comment (CoderMaggie) [inline docs/contributing/code/security.rst]:
```suggestion
    except for the vulnerabilities made public.
```

**Comment by pamil** [comment]
@CoderMaggie that's right, they're not used too much or ridiculously outdated or both.

File: docs/contributing/code/patches.rst [hunk 4]
```diff
@@ -131,10 +131,9 @@ Work on your Patch
 Work on the code as much as you want and commit as much as you want; but keep
 in mind the following:
 
-* Practice :doc:`BDD <bdd>`, which is the development methodology we use at Sylius;
+* Practice :doc:`BDD </bdd/index>`, which is the development methodology we use at Sylius;
```
Comment (CoderMaggie) [inline docs/contributing/code/patches.rst]:
```suggestion
* Practice :doc:`BDD </bdd/index>`, which is the development methodology we use in Sylius;
```
it both `at` and `in` but here we're talking about the framework, not the company 😂

**Pull request merged**